src/utils/path.js

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import type { CallNodePath } from '../types/profile-derived';
+
+export function arePathsEqual(a: CallNodePath, b: CallNodePath): boolean {
+  if (a === b) {
+    return true;
+  }
+
+  if (a.length !== b.length) {
+    return false;
+  }
+
+  // Iterating from the end because this will likely fail faster
+  for (let i = a.length - 1; i >= 0; i--) {
+    if (a[i] !== b[i]) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+// We take the easy path by converting the path to a string that will be unique.
+// In the future and if necessary we might want to have an integer-based hash so
+// that it's faster. In that case we might want to use only the n last values
+// that should be more discriminate and keep it fast.
+export function hashPath(a: CallNodePath): string {
+  return a.join('-');
+}
+
+export class PathSet implements Iterable<CallNodePath> {
```

Comment:
We should probably add some code comments about what this is.

Reply:
Oh yeah, I wanted and then forgot. Will do.

Reply:
Added as agreed on slack:
```
// This class implements all of the methods of the native Set, but provides a
// unique list of CallNodePaths. These paths can be different objects, but as
// long as they contain the same data, they are considered to be the same.
// These CallNodePaths are keyed off of the string value returned by the
// `hashPath` function above.
```